File: py/eA.py
#!python3.6
# @ guliping
# 模块事例

# 公开函数


def a():
    print("call function a")

# 私有函数


def __b():
    print("call function b")


# 公开变量
c = 10

# 私有变量
__d = 20

if __name__ == '__main__':
    print("自主运行")
else:
    print("包引入")

File: lua/study/eA.lua
#!lua 5.3.2
--@guliping
--[[
模块示例
]]

eA = {}

--定义模块变量
eA.a = 1

function eA.b()
	print("call dA.b()")
end

-- 模块私有函数
local function c()
	print("call c()")
end

--模块函数，调用模块私有函数
function eA.d()
	c()
end

--模块函数
eA.e = function()
	print("call e()")
end

return eA
